File: Puzzles/Easy/Chevaux de course/C++.cs
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

static const int MAX = 100000;
static const int MAX_PUISSANCE = 10000000;

int main()
{
    int n;
    int array[MAX];
    cin >> n; cin.ignore();
    for (int i = 0; i < n; i++) {
        cin >> array[i]; cin.ignore();
    }
    sort(array, array + n);
    int diff = MAX_PUISSANCE;
    for(int i = 0; i < n-1; i++){
        int d = array[i+1] - array[i];
        if(d < diff){
            diff = d;
        }
    }
    cout << diff  << endl;
}

File: Puzzles/Easy/May the Triforce be with you!/C++.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
  int n;
  cin >> n;
  cin.ignore();
  for (int i = 0; i < n; i++)
  {
    if (i == 0)
    {
      cout << '.';
      for (int j = 0; j < n * 2 - 2; j++)
      {
        cout << ' ';
      }
      cout << '*';
    }
    else
    {
      for (int j = 0; j < n * 2 - i - 1; j++)
      {
        cout << ' ';
      }
      for (int j = 0; j < i * 2 + 1; j++)
      {
        cout << '*';
      }
    }
    cout << '\n';
  }
  for (int i = 0; i < n; i++)
  {
    for (int j = 0; j < n - i - 1; j++)
    {
      cout << ' ';
    }
    for (int j = 0; j < i * 2 + 1; j++)
    {
      cout << '*';
    }
    for (int j = 0; j < n * 2 - (1 + (i * 2)); j++)
    {
      cout << ' ';
    }
    for (int j = 0; j < i * 2 + 1; j++)
    {
      cout << '*';
    }
    cout << '\n';
  }
}